### ISSUE TITLE
[Chia] Investigate wrong counting of missed challenges

### ISSUE THREAD

**Issue by tnt-codie** [Tea-n-Tech/chia-tea#16]
The amount of missed challenges seems wrong. It is way to many and the logfiles state that it should be fine. The whole logic needs to be revised and corrected with solid tests. It is most probably best to take logfile scenarios as basis.

**Comment by tnt-codie** [comment]
The problem is that once a missed challenge occurred there is a stau causing every following challenge to be missed too. This occurs since one list in case of multiple missed challenges is always behind the other list.

**Comment by tnt-codie** [comment]
@tnt-manu would you be so kind to document your PR changes and the inherent problems we discussed here? Thx :heart:

**Comment by tnt-manu** [comment]
Previously, the whole time logging was done with two lists on the `harvester_info` 

One list captured all the send messages, the other all the recieved messages. We then compared the last elements to see the latest time delta and put some logic around it - no recieve before send and so on.

This is now changed in PR #22 
First, I reduced the lists to variables - so `time_last_incoming_msg`  for the time of the last incoming, and `time_last_outgoing_msg` for the time of the last outgoing (farmer - harvester) message. 
An additional line_check in the log file has been introduced: `ActionFinishedSignagePoint` - this comes up on every Signage Point in the logfile of the farmer and marks our entry point for checking if one of the harvester timed_out or missed a challenge. 
So on every occurence of this finished sgn point in the log we check for each harvester if the time delta between `time_of_end_of_last_sgn_point` and `time_last_incoming_msg` is greater than a threshold (25s for missed challenge and 60s for timeout)

Why all the fuzz? It is because in the event of a cable pull on the harvester we don't get immediate disconnects and would love to see a notification on the discord. Also missed challenges are nice as well as they indicate, how good a harvester is working related to the challenges. But this is still TODO right now. The flags in the `FarmerHarvesterLogfile` (`n_overdue_responses` and `timed_out` ) are tracked with #22 but notification-wise there is no message. `n_overdue_responses` can be seen with the $farmers command and thats enough for now I guess, but time_outs currently are done by the db and the 60s metric which we know might be off as it tracks sent and recieved timestamps and in the case of a cable pull we have sends but no recieves.
